walk.go

```diff
@@ -44,71 +88,95 @@ type WalkFunc func(root, curr Node, path string, err error) error
 //     {"c":"ccc"}, // visited as foo/2
 //   ]}
 //
+// Walk returns a node constructed from the transformed result of the
+// walk function.
```

Comment:
Walk should just walk, it should not transform. I'm matching the os interface here precisely because i want to allow just walking the graph.

I understand the value of transform, but make that a separate function, namely `func Transform(root Node, TransformFunc) (Node, error) {}`.

i do not understand why `Walk` needs to transform the node. the components in the path can be escaped just fine without touching the node itself.